src/AggregateFunctions/AggregateFunctionCramersV.h

```diff
@@ -0,0 +1,121 @@
+                UInt64 count2 = count_of_value2.find(hash2)->getMapped();
+                UInt64 count_of_pair = cell.getMapped();
+                Float64 expectation = static_cast<Float64>(count1) * count2 / size_of_column;
+                chi_squared += (count_of_pair - expectation) * (count_of_pair - expectation) / expectation;
+
+            }
+            return sqrt(chi_squared / (q - 1) / size_of_column);
+        }
+
+        static constexpr UInt32 num_args = 2;
+    };
+
+
+    class AggregateFunctionCramersV :
+            public IAggregateFunctionDataHelper<AggregateFunctionCramersVData, AggregateFunctionCramersV>
+    {
+    public:
+        explicit AggregateFunctionCramersV(const DataTypes & arguments)
+                :IAggregateFunctionDataHelper<AggregateFunctionCramersVData, AggregateFunctionCramersV> ({arguments}, {})
+        {}
+
+        String getName() const override
+        {
+            return "cramersV";
+        }
+
+        bool allocatesMemoryInArena() const override { return false; } // ????
+
+        DataTypePtr getReturnType() const override
+        {
+            return std::make_shared<DataTypeNumber<Float64>>();
+        }
+
+        void add(AggregateDataPtr __restrict place, const IColumn ** columns, size_t row_num, Arena *) const override
+        {
+            ++this->data(place).size_of_column;
+            UInt64 hash1 = UniqVariadicHash<false, false>::apply(1, columns, row_num);
+            UInt64 hash2 = UniqVariadicHash<false, false>::apply(1, columns + 1, row_num);
+            UInt128 hash_pair = hash1 | (static_cast<UInt128>(hash2) << 64);
+            this->data(place).count_of_value1[hash1] += 1;
+            this->data(place).count_of_value2[hash2] += 1;
+            this->data(place).count_of_pairs[hash_pair] += 1;
+        }
+        void merge(AggregateDataPtr __restrict place, ConstAggregateDataPtr rhs, Arena *) const override
+        {
+            this->data(place).size_of_column += this->data(rhs).size_of_column;
+            for (const auto& pair : this->data(rhs).count_of_value1) {
+                UInt64 hash1 = pair.getKey();
+                UInt64 count = pair.getMapped();
+                this->data(place).count_of_value1[hash1] += count;
+            }
+            for (const auto& pair : this->data(rhs).count_of_value2) {
+                UInt64 hash2 = pair.getKey();
+                UInt64 count = pair.getMapped();
+                this->data(place).count_of_value2[hash2] += count;
+            }
+            for (const auto & pair : this->data(rhs).count_of_pairs) {
+                UInt128 hash_pair = pair.getKey();
+                UInt64 count_of_pair = pair.getMapped();
+                this->data(place).count_of_pairs[hash_pair] += count_of_pair;
+            }
+        }
+
+        void serialize(ConstAggregateDataPtr __restrict place, WriteBuffer & buf) const override
+        {
+            writeVarUInt(data(place).size_of_column, buf);
```

Comment:
You write this-> everywhere except this place, stick to the same style in one file. In my opinion, you can try to remove but I am not sure if that's possible